docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,110 @@
+as source files.
+
+Examples:
+
+```
+    csc.exe /editorconfig:.editorconfig
+    csc.exe /editorconfig:.editorconfig /editorconfig:..\.editorconfig
+    csc.exe /editorconfig:.editorconfig /editorconfig:Subdirectory\.editorconfig
+```
+
+It is expected that the compiler is given the .editorconfig files with casing that
+matches the underlying casing of the file system. The compiler *MUST NOT* do any case correction. So
+
+```
+    csc.exe /editorconfig:Subdirectory\.editorconfig /editorconfig:SubDirectory\.editorconfig
+```
+
+is a valid input to the compiler. The compiler *MAY* choose to issue a warning if two paths given
+are exactly equal by ordinal in a case-sensitive way.
+
+If multiple .editorconfigs apply to a given file (the precise definition of
+"apply" is defined later), and more than one file is defining the same value,
+.editorconfig file listed first in the command arguments takes precedence over
+later files.
+
+# Discovery of .editorconfig files in MSBuild
+
+Discovery of .editorconfig files is not done by csc.exe, but rather the host. For csproj projects,
+a new build task, DiscoverEditorConfigFiles is introduced which runs before the CoreCompile target
+at an appropriate point to locate all applicable files. The inputs to the task are:
+
+- InputFiles: a list of ITaskItems which are the input files from which we must locate .editorconfig
+  files in parent directories. The expected inputs to this include the Compile and AdditionalFile items.
+- RootDirectories: an optional list of root directories which the build task
+  *MUST NOT* ascend higher during any walking. This is a list because a build
+  environment that segregates intermediate directories or build directories
+  outside the root of the source tree might wish to ensure that no walking is
+  ever done outside the source path, or intermediate/output directories, none of which may be a
+  parent of any other root (or indeed on Windows might not even live on the same drive!)
+
+The outputs of this task are:
+
+- EditorConfigFiles: a list of ITaskItems which are paths to discovered .editorconfig files. These *MAY*
+  be relative paths to the current working directory, in the same way that input files *MAY* be relative paths.
+
+The default target maps these items to an EditorConfig item group, that is consumed by the CoreCompile target.
+
+A property DiscoverEditorConfigFiles *MUST* be provided that has the following values it can be set to:
+
+- `SearchToRoot`: the task is ran, and RootDirectories is not populated in any way.
+- `SearchToProject`: the task is ran, and RootDirectories is populated to the project directory.
+- `Off`: the task is not ran at all, via a Condition.
+
+The output of this task *MUST* be deterministic relative to the inputs. Ordering changes (even if there is no
+semantic difference) is explicitly disallowed.
+
+# PathMap
+
+*This section is informational only and does not propose any new behavior.*
```

Comment:
Doesn't that also imply that `$(TEMP)` and `$(IntermediateOutputPath)` should be a part of the default value for `$(RootDirectories)`?  If we are categorically path mapping those values then no need to search above them.